### DISCUSSION TITLE
How to query "EXISTS" instead of "IN"

### DISCUSSION THREAD

**Discussion by albertoramires** [prisma/prisma#13797] [Q&A]
Trying to do two queries similar to this:

```
where: {
  relation: { isNot: null }
}

where: {
  relation: { is: null }
}
```

The generated query uses `NOT IN` and `IN` which is incredibly slow for large tables, how can I do the same query with `NOT EXISTS` and `EXISTS` without having to use raw queries?

**Comment by austincrim** [comment]
Hey @albertoramires 👋,

It is not possible to customize this behavior without using [`$queryRaw`](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access#raw-queries-with-relational-databases).

If you are seeing reproducible performance issues with this query, please [open an issue](https://github.com/prisma/prisma/issues/new?assignees=&labels=kind%2Fbug&template=bug_report.yml) with more details so we can investigate.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
